lib/src/components/NxFileUpload/NxMultiFileUpload/NxMultiFileUpload.tsx

```diff
@@ -0,0 +1,188 @@
+        querySelectorAll<HTMLButtonElement>('.nx-selected-file__dismiss-btn') ?? []);
+    forEach((btn)=> {
+      if (selectedFile?.contains(btn) && btn === last(closeBtns)) {
+        closeBtns.length <= 1 ? inputRef.current?.focus() : closeBtns[closeBtns.length - 2].focus();
+      }
+    }, closeBtns);
+
+    const dataTransferObject = new DataTransfer();
+    if (inputRef.current) {
+      if (files) {
+        forEach((f: File) => dataTransferObject.items.add(f), without([fileObj], Array.from(files)));
+        inputRef.current.files = dataTransferObject.files;
+
+        const normalizedFiles = !inputRef.current.files.length ? null : inputRef.current.files;
+        onChangeProp(normalizedFiles);
+      }
+    }
+  }
+
+  // the container scrolls to the bottommost file only when new files are added
+  useEffect(function() {
+    const newFilesAdded = previousFiles === null || (files && (files.length > previousFiles.length));
+
+    if (selectedFilesContainerRef.current && newFilesAdded && files) {
+      const lastSelectedFile = selectedFilesContainerRef.current.lastElementChild;
+      lastSelectedFile?.scrollIntoView({block: 'nearest'});
+    }
+    if (files) {
+      setPreviousFiles(files);
+    }
+  }, [files]);
+
+  return (
+    <div ref={ref} className={className} >
+      <div className="nx-multi-file-upload__container">
+        <input ref={inputRef}
+               { ...attrs }
+               disabled={disabled}
+               onChange={onChange}
+               id={inputId}
+               className="nx-multi-file-upload__input"
+               type="file"
+               aria-label={isFileSelected ? `${files.length} files selected` : 'No File Selected'}
+               aria-required={isRequired ?? undefined}
+               aria-invalid={showError || undefined}
+               aria-errormessage={showError ? validationErrorId : undefined}
+               multiple/>
+        {/* Keynav and screenreaders can ignore the button itself in favor of the <input> */}
+        <NxButton type="button"
+                  aria-hidden={true}
+                  disabled={disabled}
+                  variant="tertiary"
+                  onClick={openPicker}
+                  tabIndex={-1}
+                  className="nx-multi-file-upload__select-btn">
+          Add Files
+        </NxButton>
+        <div ref={selectedFilesContainerRef} className="nx-multi-file-upload__container__files nx-scrollable">
+          { isFileSelected ?
+            Array.from(files).map((file, key) =>
+              <SelectedFile key= {`${key}__selected-file}`} file={file} onDismiss={onDismiss}/>
+            ) :
+            <span className={noFileMessageClassName}>
+              <span>No file selected</span>
+              <NxFontAwesomeIcon icon={faExclamationCircle} />
+            </span>
+          }
+        </div>
+      </div>
+
+      { showError &&
+        // TODO confirm whether this text should be hard-coded
```

Comment:
This TODO text should have been caught and removed from `NxFileUpload` during its original PR.  Definitely don't copy it into yours

Reply:
> This TODO text should have been caught and removed from `NxFileUpload` during its original PR. Definitely don't copy it into yours

Thanks for clarifying, removed: [915d502](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/915d502b5258f819830445e0ee5515c487b5dc52)